src/Kiota.Builder/Extensions/OpenApiSchemaExtensions.cs

```diff
@@ -30,44 +30,43 @@ public static IEnumerable<string> GetSchemaNames(this OpenApiSchema schema)
         return Enumerable.Empty<string>();
     }
     private static IEnumerable<string> FlattenIfRequired(this IList<OpenApiSchema> schemas, Func<OpenApiSchema, IList<OpenApiSchema>> subsequentGetter) {
-        var resultSet = schemas;
-        if(schemas.Count == 1 && string.IsNullOrEmpty(schemas.First().Title))
-            resultSet = schemas.FlattenEmptyEntries(subsequentGetter, 1);
-        
-        return resultSet.Select(x => x.Title).Where(x => !string.IsNullOrEmpty(x));
+        return (schemas.Count == 1 && string.IsNullOrEmpty(schemas.First().Title) ?
+                    schemas.FlattenEmptyEntries(subsequentGetter, 1) :
+                    schemas)
+            .Select(static x => x.Title).Where(static x => !string.IsNullOrEmpty(x));
     }
 
     public static string GetSchemaName(this OpenApiSchema schema) {
-        return schema.GetSchemaNames().LastOrDefault()?.TrimStart('$');// OData $ref
+        return schema.GetSchemaNames().LastOrDefault()?.TrimStart('$') ?? string.Empty;// OData $ref
     }
 
     public static bool IsReferencedSchema(this OpenApiSchema schema) {
         var isReference = schema?.Reference != null;
-        if(isReference && schema.Reference.IsExternal)
+        if(isReference && schema!.Reference.IsExternal)
```

Comment:
schema isn't nullable here. Is there a reason it's accessed using `schema?.<...>` then `schema!.<...>`?

Reply:
https://github.com/microsoft/kiota/blob/feature/linting/tests/Kiota.Builder.Tests/Extensions/OpenApiSchemaExtensionsTests.cs#L40 a consuming library without NRT enabled can still pass null since it's a public method.

Reply:
Maybe we should also add a `[NotNull]` attribute

Reply:
yeah I'm not entirely sure how much of an experience difference this attribute makes vs not having the ? I guess we'll get feedback about that as people consume the library.